File: TypeScript/Practice/LeetCode/Easy/Sort the People.ts
// https://leetcode.com/problems/sort-the-people/

function sortPeople(names: string[], heights: number[]): string[] {
    const namesAndHeights: [string, number][] = [];
    const sortedPeople: string[] = [];

    for (let i = 0; i < names.length; i++) {
        namesAndHeights.push([names[i], heights[i]]);
    }

    namesAndHeights.sort((a, b) => b[1] - a[1]);

    for (let i = 0; i < namesAndHeights.length; i++) {
        sortedPeople.push(namesAndHeights[i][0]);
    }
    
    return sortedPeople;
};

File: C#/Practice/LeetCode/Easy/Sort the People.cs
// https://leetcode.com/problems/sort-the-people/

public class Solution {
    public string[] SortPeople(string[] names, int[] heights) {
        List<Tuple<string, int>> namesAndHeights = new List<Tuple<string, int>>();
        List<string> sortedPeople = new List<string>();

        for (int i = 0; i < names.Length; i++)
        {
            namesAndHeights.Add(Tuple.Create(names[i], heights[i]));
        }

        namesAndHeights = namesAndHeights.OrderByDescending(t => t.Item2).ToList();

        for (int i = 0; i < namesAndHeights.Count; i++)
        {
            sortedPeople.Add(namesAndHeights[i].Item1);
        }

        return sortedPeople.ToArray();
    }
}